core/src/test/java/org/apache/iceberg/TestManifestWriterVersions.java

```diff
@@ -213,27 +224,125 @@ public void testV2ManifestRewriteWithInheritance() throws IOException {
     checkRewrittenEntry(readManifest(manifest2), 0L, FileContent.DATA);
   }
 
+  @Test
+  public void testV3Write() throws IOException {
+    ManifestFile manifest = writeManifest(3);
+    checkEntry(
+        readManifest(manifest),
+        ManifestWriter.UNASSIGNED_SEQ,
+        ManifestWriter.UNASSIGNED_SEQ,
+        FileContent.DATA,
+        FIRST_ROW_ID);
+  }
+
+  @Test
+  public void testV3WriteWithInheritance() throws IOException {
+    DataFile withoutFirstRowId =
+        DataFiles.builder(SPEC).copy(DATA_FILE).withFirstRowId(null).build();
+
+    ManifestFile manifest = writeAndReadManifestList(writeManifest(3, withoutFirstRowId), 3);
+    checkManifest(manifest, SEQUENCE_NUMBER);
+    assertThat(manifest.content()).isEqualTo(ManifestContent.DATA);
+
+    // v2+ should use the correct sequence number by inheriting it
```

Comment:
Not sure follow the V2 comment here, is this just to point out we should be inheriting both sequence and row id?

Reply:
This is saying that the sequence number behavior should match v2.

Reply:
Ah got it!